### PR TITLE
Update redirect docs to be clearer

### PR DESCRIPTION
**Pull request by markstory**
* Add more information on doing redirects in event handlers
* Simplify examples and reduce verbosity.

Refs cakephp/cakephp#11197

### CODE REVIEW

File: en/controllers.rst [hunk 1]
```diff
@@ -289,70 +289,53 @@ Redirecting to Other Pages
...
+
+    return $this->redirect($this->referer());
+
+By using the second parameter you can define a status code for your redirect::
+
+    // Do a 301 (moved permanently)
+    return $this->redirect('/order/confirm', 301);
+
+    // Do a 303 (see other)
+    return $this->redirect('/order/confirm', 301);
```
Comment (othercorey) [inline en/controllers.rst]:
```suggestion
    return $this->redirect('/order/confirm', 303);
```

File: en/controllers/components.rst [hunk 2]
```diff
@@ -311,6 +306,23 @@ augment the request cycle.
...
+.. _redirect-component-events:
+
+Using Redirects in Component Events
+===================================
+
+To redirect from within a component callback method you can use the following::
+
+    public function beforeFilter(EventInterface $event)
+    {
+      $event->stopPropagation();
```
Comment (othercorey) [inline en/controllers/components.rst]:
```suggestion
        $event->stopPropagation();
```

File: en/controllers/components.rst [hunk 3]
```diff
@@ -311,6 +306,23 @@ augment the request cycle.
...
+      return $this->getController()->redirect('/');
```
Comment (othercorey) [inline en/controllers/components.rst]:
```suggestion
        return $this->getController()->redirect('/');
```

**Comment by markstory** [comment]
Thank you for fixing my stupid mistakes @othercorey 🙇

**Pull request merged**